pallets/name-service/src/tests.rs

```diff
@@ -0,0 +1,297 @@
+
+use super::*;
+use crate::mock::{NameService, RuntimeOrigin as MockOrigin, *};
+use frame_support::{assert_noop, assert_ok};
+
+pub const ALICE: sp_runtime::AccountId32 = sp_runtime::AccountId32::new([0u8; 32]);
+pub const BOB: sp_runtime::AccountId32 = sp_runtime::AccountId32::new([1u8; 32]);
+
+/// Initializes a test by funding accounts.
+#[inline]
+fn initialize_test() {
+    assert_ok!(Balances::set_balance(
+        MockOrigin::root(),
+        ALICE,
+        1_000_000_000_000_000,
+        0
+    ));
+    assert_ok!(Balances::set_balance(
+        MockOrigin::root(),
+        BOB,
+        1_000_000_000_000_000,
+        0
+    ));
+}
+
+#[test]
+fn register_should_work() {
+    ExtBuilder::default().build().execute_with(|| {
+        initialize_test();
+        assert_ok!(NameService::register(
+            MockOrigin::signed(ALICE),
+            "test".as_bytes().to_vec(),
+            ALICE.into()
+        ));
+        System::set_block_number(5);
+        assert_ok!(NameService::accept_register(
+            MockOrigin::signed(ALICE),
+            "test".as_bytes().to_vec(),
+            ALICE.into()
+        ));
+    });
+}
+
+#[test]
+fn set_primary_should_work() {
+    ExtBuilder::default().build().execute_with(|| {
+        initialize_test();
+        assert_ok!(NameService::register(
+            MockOrigin::signed(ALICE),
+            "test".as_bytes().to_vec(),
+            ALICE.into()
+        ));
+        System::set_block_number(5);
+        assert_ok!(NameService::accept_register(
+            MockOrigin::signed(ALICE),
+            "test".as_bytes().to_vec(),
+            ALICE.into()
+        ));
+        assert_ok!(NameService::set_primary_name(
+            MockOrigin::signed(ALICE),
+            "test".as_bytes().to_vec(),
+            ALICE.into()
+        ));
+    });
+}
+
+#[test]
+fn cancel_register_should_work() {
+    ExtBuilder::default().build().execute_with(|| {
+        initialize_test();
+        assert_ok!(NameService::register(
+            MockOrigin::signed(ALICE),
+            "test".as_bytes().to_vec(),
+            ALICE.into()
+        ));
+        assert_ok!(NameService::cancel_pending_register(
+            MockOrigin::signed(ALICE),
+            "test".as_bytes().to_vec(),
+            ALICE.into()
+        ));
+    });
+}
+
+#[test]
+fn remove_register_should_work() {
+    ExtBuilder::default().build().execute_with(|| {
+        initialize_test();
+        assert_ok!(NameService::register(
+            MockOrigin::signed(ALICE),
+            "test".as_bytes().to_vec(),
+            ALICE.into()
+        ));
+        System::set_block_number(5);
+        assert_ok!(NameService::accept_register(
+            MockOrigin::signed(ALICE),
+            "test".as_bytes().to_vec(),
+            ALICE.into()
+        ));
+        assert_ok!(NameService::remove_register(
+            MockOrigin::signed(ALICE),
+            "test".as_bytes().to_vec(),
+            ALICE.into()
+        ));
+    });
+}
```

Comment:
can you add a check to re-register a new name after de-registering the first one
I would also assert the storage items values just in case.

Reply:
Also canceling and re-registering